pkg/applicationserver/io/mqtt/mqtt.go

```diff
@@ -315,6 +367,18 @@ func (c *connection) CanWrite(info *auth.Info, topicParts ...string) bool {
 
 func (c *connection) deliver(pkt *packet.PublishPacket) {
 	logger := log.FromContext(c.io.Context()).WithField("topic", pkt.TopicName)
+
+	if c.limit() {
+		// TODO: observability
+		err := errTrafficRateLimitExceeded.WithAttributes(
+			"application_uid", unique.ID(c.io.Context(), c.io.ApplicationIDs()),
+			"remote_addr", c.mqtt.RemoteAddr().String(),
+		)
+		logger.WithError(err).Error("Terminating connection")
```

Comment:
This shouldn't be an `Error` log; those are reserved for things that shouldn't happen (possible bugs). A `Warn` log is better here.

Reply:
Also it should be imperative